Syntax for comments

akanellop opened the pull request.

akanellop committed:
implementation of the comment feature
-check if there is a # symbol and parse the line accordingly
-filter the entries for empty ones created by a comment
-added the render for comments ,same code for all features

akanellop committed:
used pretty for better formation

parisk commented on src/main.js:
```diff
@@ -110,28 +112,44 @@ class Entry {
-      //if there is a default value indicated by an existing "="
-      if (element[1].includes("=")) {
-        element[1] = element[1].split("=");
+  return (envSpecAsArrayParsed = envSpecAsArray
```
We do not need the assignment here.  Just `return envSpecAsArray.map(...`

akanellop committed:
Suggestion : added this check to on the validation function so if an error occured ,we wouldnt have to check the rest of the document

parisk commented on src/main.js:
```diff
@@ -94,12 +94,14 @@ class Entry {
+   *@param {string} envComment given comment using the # symbol
    */
-  constructor(envName, envType, envChoices, envDefaultVal) {
+  constructor(envName, envType, envChoices, envDefaultVal, envComment) {
```
I think we can rename all arguments to simpler names 😄. What about just `name`, `type`, `choices`, `defaultValue` and `comment`?

The `env...` prefix does not provide anything useful to the reader.

**P.S.:** Do not forget to update the JSDoc as well

parisk commented on src/main.js:
```diff
@@ -94,12 +96,14 @@ class Entry {
     this.type = envType;
     this.choices = envChoices;
     this.defaultValue = envDefaultVal;
+    this.comment = envComment;
```
I suggest [`trim`ing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trim) the comment. There is no reason to maintain leading and trailing white spaces.

parisk commented on src/main.js:
```diff
@@ -110,28 +114,44 @@ class Entry {
+        return new Entry(element.trim(), "text", null, null, comment);
+      }
+    })
+    .filter(element => element.name !== "")); //in case a line started with "#" we would have a blank environmental variable
```
It's better to move the filtering before `map`. There is no reason to run operations in `map` for lines that will be ignored.

Let's just filter out lines starting with `#` and then apply `map` to the rest.

akanellop committed:
changes
filter function separated from map in parseVarFromType function
better variable names

parisk commented on src/main.js:
```diff
@@ -110,7 +114,17 @@ class Entry {
 const parseVarFromType = envSpecAsArray => {
-  return (envSpecAsArrayParsed = envSpecAsArray.map(element => {
+  //in case a line started with "#" we would have a blank environmental variable
+  envSpecAsArray = envSpecAsArray.filter(line => line[0] !== "#");
```
First, we should avoid reassigning values to arguments inside a function. You can just:

```javascript
return envSpecAsArray.filter(...).map(...)
```

Next, we can just use the [`startswith` JavaScript function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith) instead of this manual check 😄.

parisk commented on src/main.js:
```diff
@@ -90,16 +92,18 @@ const checkValidationOfValues = envSpecString => {
-    this.type = envType;
-    this.choices = envChoices;
-    this.defaultValue = envDefaultVal;
+  constructor(name, type, choices, defaultVal, comment) {
```
Rename `defaultVal` to `defaultValue` please. There is no reason to abbreviate it (update JSDoc as well).

parisk commented on src/main.js:
```diff
@@ -110,7 +114,17 @@ class Entry {
+  envSpecAsArray = envSpecAsArray.filter(line => line[0] !== "#");
+
+  return envSpecAsArray.map(element => {
+    comment = "";
```
Instead of initialising this with an empty value and using if statements below, we can use regular expressions instead.

Something like that could make the cut:

```javascript
const commentRegexp = /#(.+)$/;
const commentMatch = element.match(commentRegexp);
const comment = (commentMatch) ? commentMatch[1] : null;
```

Or something like that 😄

parisk commented on src/main.js:
```diff
@@ -181,9 +201,12 @@ const outputHTML = envSpecEntriesArray => {
-        return toPrint;
       }
+      //in case of existing comment add it to toPrint value
+      if (element.comment !== "") {
```
This can be simplified to:

```javascript

if (element.comment) {
  // ...
}
```

akanellop committed:
change syntax implementation for comments

parisk reviewed (approved):
Great, thanks!
